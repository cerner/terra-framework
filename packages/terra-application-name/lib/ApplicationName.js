"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ApplicationHeaderName", {
  enumerable: true,
  get: function get() {
    return _ApplicationHeaderName.default;
  }
});
Object.defineProperty(exports, "ApplicationMenuName", {
  enumerable: true,
  get: function get() {
    return _ApplicationMenuName.default;
  }
});
exports.default = void 0;
var _ApplicationHeaderName = _interopRequireDefault(require("./ApplicationHeaderName"));
var _ApplicationMenuName = _interopRequireDefault(require("./ApplicationMenuName"));
var ApplicationName = {
  ApplicationHeaderName: _ApplicationHeaderName.default,
  ApplicationMenuName: _ApplicationMenuName.default
};
var _default = ApplicationName;
exports.default = _default;