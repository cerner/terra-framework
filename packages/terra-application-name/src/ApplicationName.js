import ApplicationHeaderName from './ApplicationHeaderName';
import ApplicationMenuName from './ApplicationMenuName';

const ApplicationName = {
  ApplicationHeaderName,
  ApplicationMenuName,
};

export default ApplicationName;
export { ApplicationHeaderName, ApplicationMenuName };
