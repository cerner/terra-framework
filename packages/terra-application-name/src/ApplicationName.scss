:local {
  .application-name {
    /* TODO: Add styles */
  }
}
