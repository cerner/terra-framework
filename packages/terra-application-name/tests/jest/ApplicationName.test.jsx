import React from 'react';
import ApplicationName from '../../src/ApplicationName';

describe('ApplicationName', () => {
  const defaultRender = <ApplicationName />;

  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper).toMatchSnapshot();
  });

  // Prop Tests
  it('should use the default value when no value is given', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.find('.application-name').text()).toEqual('default');
  });

  // Structure Tests
  it('should have the class application-name', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper.prop('className')).toContain('application-name');
  });
});
