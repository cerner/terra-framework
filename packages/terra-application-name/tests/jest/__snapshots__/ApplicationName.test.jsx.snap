// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationName should render a default component 1`] = `
<div
  className="application-name"
>
  default
</div>
`;
